fenix/app/src/main/java/org/mozilla/fenix/search/awesomebar/AwesomeBarView.kt

```diff
@@ -278,45 +283,77 @@ class AwesomeBarView(
-                providersToAdd.add(defaultCombinedHistoryProvider)
-            } else {
-                providersToAdd.add(defaultHistoryStorageProvider)
-            }
+            providersToAdd.add(
+                getHistoryProvider(
+                    filters = listOfNotNull(getFilterToExcludeSponsoredSuggestions(state)),
+                ),
+            )
         }
 
         if (state.showHistorySuggestionsForCurrentEngine) {
-            getHistoryProvidersForSearchEngine(state.searchEngineSource)?.let {
-                providersToAdd.add(it)
-            }
+            providersToAdd.add(
+                getHistoryProvider(
+                    filters = listOfNotNull(
+                        getFilterToExcludeSponsoredSuggestions(state),
+                        getFilterForCurrentEngine(state),
+                    ),
+                ),
+            )
         }
 
         if (state.showAllBookmarkSuggestions) {
-            providersToAdd.add(getBookmarksProvider(state.searchEngineSource))
+            providersToAdd.add(
+                getBookmarksProvider(
+                    filters = listOfNotNull(getFilterToExcludeSponsoredSuggestions(state)),
+                ),
+            )
         }
 
         if (state.showBookmarksSuggestionsForCurrentEngine) {
-            providersToAdd.add(getBookmarksProvider(state.searchEngineSource, true))
+            providersToAdd.add(
+                getBookmarksProvider(
+                    filters = listOfNotNull(
+                        getFilterToExcludeSponsoredSuggestions(state),
+                        getFilterForCurrentEngine(state),
+                    ),
+                ),
+            )
         }
 
         if (state.showSearchSuggestions) {
             providersToAdd.addAll(getSelectedSearchSuggestionProvider(state))
         }
 
         if (state.showAllSyncedTabsSuggestions) {
-            providersToAdd.add(getSyncedTabsProvider(state.searchEngineSource))
+            providersToAdd.add(
+                getSyncedTabsProvider(
+                    filters = listOfNotNull(getFilterToExcludeSponsoredSuggestions(state)),
+                ),
+            )
         }
 
         if (state.showSyncedTabsSuggestionsForCurrentEngine) {
-            providersToAdd.add(getSyncedTabsProvider(state.searchEngineSource, true))
+            providersToAdd.add(
+                getSyncedTabsProvider(
+                    filters = listOfNotNull(
+                        getFilterToExcludeSponsoredSuggestions(state),
+                        getFilterForCurrentEngine(state),
+                    ),
+                ),
+            )
         }
 
         if (activity.browsingModeManager.mode == BrowsingMode.Normal && state.showAllSessionSuggestions) {
-            providersToAdd.add(getLocalTabsProvider(state.searchEngineSource))
+            // Unlike other providers, we don't exclude sponsored results for open tabs.
```

Comment:
This matches the Desktop behavior that I observed: Desktop hides sponsored suggestions from history, bookmarks, and synced tabs, but not open local tabs.